demystifying_large_language_models/demystifying_large_language_models.md

```diff
@@ -0,0 +1,669 @@
+
+![A figure with a hat states, "Oh, hey, you organized our photo archive!".  A second figure replies, "Yeah, I trained a neural net to sort the unlabeled photos into categories."  The first figure resonds, "Whoa! Nice work!."  The caption below the drawing reads: "Engineering tip: When you do a task by hand, you can technically say you trained a neural net to do it."](https://imgs.xkcd.com/comics/trained_a_neural_net.png "[Trained a Neural Net](https://xkcd.com/2173/) comic by xkcd, [CC BY-NC 2.5](https://xkcd.com/license.html).")<!-- style = "max-width: 220px;" -->Neural networks take this power of trial and error and massively expand it. Neural networks are an attempt to in some way replicate the operations of neurons in a brain. Various nodes (neurons) receive input, perform some mathematical transformation on that input, and either "fire" (pass some output on to a next step) or not, depending on whether the result of the calculation passes a certain threshold.  Each node starts off with random weights that determine how to mathematically transform the data it receives, and the computer tries many combinations of weights to see which ones are the most accurate. 
+
+
+An Example
+-----
+
+Let's consider a binary question.  Will a given person buy takeout for dinner?  Given seven input variables, you want a single outcome: 1 for "yes, they will get takeout" and 0 for "no, they will not get takeout."
+
+In this image (created thanks to [NN-SVG](http://alexlenail.me/NN-SVG/index.html)), the leftmost column of circles represent **inputs**.  Maybe one input is an individual's level of tiredness, another is the number of days they have had takeout in the last week, a third input could be the number of leftovers in the person's fridge.  These inputs are scaled and normalized.
+
+![A depiction of a neural net, which consists of three columns of circles, the first of which has seven circles and is the input layer, the second of which has ten circles and is the hidden layer, and the last of which has one circle and is the output layer.  Lines connect the circles from the first column to circles in the second column, and circles in the second column to the circle in the third column.](media/example_nn.png)Lines of various intensity show the **weights** (think of them like coefficients in an equation) that each of these inputs contribute to ten nodes in the **hidden layer**, which is a set of nodes between input and output. The darker the line, the heavier that input is weighted for that node's calculation. Some nodes might heavily consider one's previous takeout habits, for example, while others disregard it entirely. Each hidden layer node calculates the sum of all its weighted inputs, and each one comes up with its own unique answer.<br/><br/>
```

Comment:
```suggestion
![A depiction of a neural net, which consists of three columns of circles, representing the input, hidden, and output layers. The input layer consists of seven circles, the hidden layer has ten circles, and the output layer has just one circle. Each circle of the input layer is connected by a line to each of the circles in the hidden layer, creating a large, overlapping web of connections. The lines are of varying darkness, with some so faint they are almost invisible. Then, each circle in the hidden layer has just one line connecting it to the output layer's single circle. These lines are also of varying darkness.](media/example_nn.png)Lines of various intensity show the **weights** (think of them like coefficients in an equation) that each of these inputs contribute to ten nodes in the **hidden layer**, which is a set of nodes between input and output. The darker the line, the heavier that input is weighted for that node's calculation. Some nodes might heavily consider one's previous takeout habits, for example, while others disregard it entirely. Each hidden layer node calculates the sum of all its weighted inputs, and each one comes up with its own unique answer.<br/><br/>
```

added some more contextualizing info about the visual characteristics of the very many lines between columns 1 & 2, and clean single line between 2 & 3, which I think is an important part of the visual representation here.